### PR TITLE
Add rule S3854 (based on 'constructor-super') for TS

### LINKED ISSUE #1458: Add rule S3854 (based on 'constructor-super') for TS

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #1458 

No issues on ruling (which is ok as this problem produces runtime error)

### CODE REVIEW

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S3854.html [hunk 1]
```diff
@@ -12,82 +12,23 @@ <h3>Known Limitations</h3>
 </ul>
 <h2>Noncompliant Code Example</h2>
 <pre>
-class Animal {
```
Comment (vilchik-elena) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S3854.html]: I removed a lot of examples as they don't make sense for TS.

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/ExpectedIssues.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    String ruleKey = file.toFile().getName().split("\\.")[0];
+    String fileKey = projectKey + ":" + projectDirPath.relativize(file).toString().replaceAll(Pattern.quote(File.separator), "/");
+    List<String> lines = getLines(file);
+    for (int lineNumber = 1; lineNumber <= lines.size(); lineNumber++) {
+      String currentLine = lines.get(lineNumber - 1);
+
+      if (!NON_COMPLIANT_PATTERN.matcher(currentLine).find() || currentLine.trim().startsWith("//")) {
+        // continue if no "// Noncompliant" in the line,
+        // or the entire line is commented (to not consider commented out code)
+        continue;
```
Comment (quentin-jaquier-sonarsource) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/ExpectedIssues.java]: Do you need this continue? Why not inverting the condition of the `if` and put line 77 inside the body?

Comment (vilchik-elena) [reply its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/ExpectedIssues.java]: nice idea for a rule:)

Comment (vilchik-elena) [reply its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/ExpectedIssues.java]: I will merge as it is as I plan to improve this test further.

**Pull request merged**